clap_derive/src/derives/arg_enum.rs

```diff
@@ -116,14 +116,8 @@ fn gen_from_str(lits: &[(TokenStream, Ident)]) -> TokenStream {
 
     quote! {
         fn from_str(input: &str, case_insensitive: bool) -> ::std::result::Result<Self, String> {
-            let func = if case_insensitive {
-                ::std::ascii::AsciiExt::eq_ignore_ascii_case
-            } else {
-                str::eq
-            };
-
             match input {
-                #(val if func(val, #lit) => Ok(Self::#variant),)*
+                #(val if #lit.matches(val, case_insensitive) => Ok(Self::#variant),)*
                 e => Err(format!("Invalid variant: {}", e)),
             }
         }
```

Comment:
Can and should we make `from_str` an inherent method?  It looks like we could loop over `Self::variants()` and I don't think it would make much of a difference, performance wise.

Reply:
We could, but the ArgValue has no idea of the enum variant. Or what did you mean?
We could make variants return a `Vec<(ArgValue, ArgEnum)>`.

Reply:
You're right.  For this to work, we'd need an iterator over the variants that we could then call `as_arg` on.

On one hand, it adds a function to the trait that is only used for implementing the inherent method.  On the other hand, it simplifies the trait for people manually implementing it (not common but possible, I have a PR doing just that)

Overall I can go either way